src/SDKs/SqlManagement/Sql.Tests/Utilities/SqlManagementTestUtilities.cs

```diff
@@ -439,5 +408,24 @@ internal static string GetServerKeyNameFromKeyBundle(KeyBundle keyBundle)
             string vaultName = keyBundle.KeyIdentifier.VaultWithoutScheme.Split('.').First();
             return $"{vaultName}_{keyBundle.KeyIdentifier.Name}_{keyBundle.KeyIdentifier.Version}";
         }
+
+        public static void ExecuteWithRetry(System.Action action, TimeSpan timeout, TimeSpan RecordRetryDelay, Func<CloudException, bool> acceptedErrorFunction)
```

Comment:
@nathannfan 
Couple of things you might want to consider here:
1) RecordRetryDelay is completely getting ignore
Consider not hardcoding retryDelay to 5 seconds, rather initialize it to the supplied argument (RecordRetryDelay)
2) Consider using condition (DateTime.UtcNow < timeout) on the while part rather while(true)
3) Correct me if I understand your entire retry logic:
You do retry on CloudException, but what if your Action Delegate always throw, so on your last exception before the timeout you will end up swalloing the exception and will never be bubbled up.
In one of your usage, Assume your CreateOrUpdate fails, you will end up retrying 24 times (assuming your CreateOrUpdate takes less than 5 seconds to complete)

Consider modelling your retry condition on number of retries rather retryDelay.
So in the above case, if your CreateOrUpdate fails for 3 consecutive times, do you anticipate it to succeed the fourth time?
if not then what's the point in retrying after 3rd failure?

If your logic is based on number of retries, you are automatically saved from a secanario where user unknowingly provides 30 mins as timeout and RecrodRetryDelay as 3 seconds.

4) use Microsoft.Rest.ClientRuntime.Azure.TestFramework.TestUtilities.Wait() instead of Thread.Sleep
5) nit: feels odd to have a return in a void function.
Consider a boolean that will help you to break out of while loop on success (non-exception scenario)